include/inviwo/core/datastructures/geometry/typedmesh.h

```diff
@@ -0,0 +1,489 @@
+    std::shared_ptr<Buffer<type>> getEditableTexCoords() { return Base::buffer_; }
+
+    void setVertexTexCoord(size_t index, vec3 texCoord){
+        getEditableTexCoords()->getEditableRAMRepresentation()->set(index,texCoord);
+    }
+};
+
+/**
+ * \ingroup typedmesh
+ * BufferTrait for Curvature buffers (float)
+ */
+class CurvatureBuffer
+    : public BufferTrait<float, 1, static_cast<int>(BufferType::CurvatureAttrib)> {
+public:
+    using Base = BufferTrait<float, 1, static_cast<int>(BufferType::CurvatureAttrib)>;
+    using Base::Base;
+
+    std::shared_ptr<const Buffer<type>> getCurvatures() const { return Base::buffer_; }
+    std::shared_ptr<Buffer<type>> getEditableCurvatures() { return Base::buffer_; }
+
+    void setVertexCurvature(size_t index, float curvature){
+        getEditableCurvatures()->getEditableRAMRepresentation()->set(index,curvature);
+    }
+};
+
+/**
+ * \ingroup typedmesh
+ * BufferTrait for Uint32 buffers
+ */
+using IndexBuffer = BufferTrait<uint32_t, 1, static_cast<int>(BufferType::IndexAttrib)>;
+
+/**
+ * \ingroup typedmesh
+ * BufferTrait for radii buffers (float)
+ * \see SphereMesh
+ */
+class RadiiBuffer
+    : public BufferTrait<float, 1, static_cast<int>(BufferType::NumberOfBufferTypes)> {
+public:
+    using Base = BufferTrait<float, 1, static_cast<int>(BufferType::NumberOfBufferTypes)>;
+    using Base::Base;
+
+    std::shared_ptr<const Buffer<type>> getRadii() const { return Base::buffer_; }
+    std::shared_ptr<Buffer<type>> getEditableRadii() { return Base::buffer_; }
+
+    void setVertexRadius(size_t index, float radius){
+        getEditableRadii()->getEditableRAMRepresentation()->set(index,radius);
+    }
+};
+}  // namespace buffertraits
+
+/**
+ * \class TypedMesh
+ * \ingroup datastructures
+ * \ingroup typedmesh
+ *
+ *
+ *  TypedMesh is a templated data structure for creating meshes with a custom amount of vertex
+ * buffers. It uses a variadic set of BufferTraits to define its interface. For example, a Mesh with
+ * a position and color per vertex would be defined as
+ * `TypedMesh<PositionsBufferTrait,ColorsBufferTrait>`. Depending on the Traits specified in the
+ * declaration the interface towards the class is updated.
+ *
+ * # Simple Usage
+ *
+ * \code{.cpp}
+ * using MyMesh = TypedMesh<buffertraits::PositionsBuffer,buffertraits::ColorsBuffer>;
+ * MyMesh mesh;
+ * mesh.addVertex(vec3(0.0f), vec4(1,0,0,1) );
+ * mesh.addVertex(vec3(1.0f), vec4(0,1,0,1) );
+ * \endcode
+ *
+ * If texture coordinates is also needed for each vertex in the mesh then one could instead use:
+ *
+ * \code{.cpp}
+ * using MyMesh = TypedMesh<buffertraits::PositionsBuffer, buffertraits::TexcoordBuffer,
+ * buffertraits::ColorsBuffer>; MyMesh mesh; mesh.addVertex(vec3(0.0f), vec3(0.0f), vec4(1,0,0,1) );
+ * mesh.addVertex(vec3(1.0f), vec3(1.0f), vec4(0,1,0,1) );
+ * \endcode
+ *
+ * For meshes when more than a few vertices is added is more efficient to add all vertices to the
```

Comment:
For meshes with more than a couple of vertices, it is more efficient to add all vertices to the ...